Anki.Vector/Robot.cs

```diff
@@ -284,8 +284,19 @@ internal void PropagateException(Exception ex)
         /// <returns>A task that represents the asynchronous operation; the task result contains the connected robot instance</returns>
         public static async Task<Robot> NewConnection(IRobotConfiguration robotConfiguration, int timeout = DefaultConnectionTimeout)
         {
+            // Create a robot proxy object used to communicate with Vector
             var robot = new Robot();
-            await robot.Connect(robotConfiguration, timeout).ConfigureAwait(false);
+            try
+            {
+                // Attempt to connect the robot
+                await robot.Connect(robotConfiguration, timeout).ConfigureAwait(false);
+            }
+            catch(Exception e)
+            {
+                // There was a problem connecting with the robot, clean up the connection
+                robot.Dispose();
+                throw e;

```

Comment:
Hint for .NET: `throw e;` and `throw; ` do different things.  The former will throw away the previous stack trace and the latter will keep the original.  So in this case, code it should be:

            catch(Exception)
            {
                // There was a problem connecting with the robot, clean up the connection
                robot.Dispose();
                throw;
             }

I could have made this change myself after the merge but I thought I'd mention it if you didn't know already.